builder/dockerfile/internals_windows.go

```diff
@@ -0,0 +1,392 @@
+
+	var subKeyCount uint32
+	var maxSubKeyLen uint32
+	var accountName string
+
+	err := windows.RegQueryInfoKey(groupNamesKey, nil, nil, nil, &subKeyCount, &maxSubKeyLen, nil, nil, nil, nil, nil, nil)
+	if err != nil {
+		return "", false, err
+	}
+
+	accountLocated := false
+	accountBuffer := make([]uint16, maxSubKeyLen+1)
+	maximumAccountBufferLength := uint32(len(accountBuffer))
+	accountBufferLength := maximumAccountBufferLength
+	accountSid := ""
+
+	for index := uint32(0); index < subKeyCount; index++ {
+		accountBufferLength = maximumAccountBufferLength
+		err := windows.RegEnumKeyEx(groupNamesKey, index, &accountBuffer[0], &accountBufferLength, nil, nil, nil, nil)
+
+		if err != nil {
+			return "", false, err
+		}
+
+		accountName = windows.UTF16ToString(accountBuffer[:])
+
+		err = changeKeySecurity(groupNamesKey, accountName)
+		if err != nil {
+			return "", false, err
+		}
+
+		if accountName == groupStr {
+			accountLocated = true
+			break
+		}
+	}
+
+	if accountLocated {
+		var accountKey windows.Handle
+
+		err := windows.RegOpenKeyEx(groupNamesKey, &accountBuffer[0], 0, windows.KEY_READ, &accountKey)
+		if err != nil {
+			return "", false, err
+		}
+
+		var groupRid uint32
+
+		err = windows.RegQueryValueEx(accountKey, nil, nil, &groupRid, nil, nil)
+		if err != nil {
+			return "", false, err
+		}
+
+		accountSid = computerSid + "-" + strconv.FormatUint(uint64(groupRid), 10)
+
+		defer windows.RegCloseKey(accountKey)
+	}
+
+	return accountSid, accountLocated, nil
+}
+
+func changeKeySecurity(rootKey windows.Handle, subkey string) error {
+	var innerKey windows.Handle
+
+	err := windows.RegOpenKeyEx(rootKey, windows.StringToUTF16Ptr(subkey), 0, system.READ_CONTROL|system.WRITE_DAC, &innerKey)
+	if err != nil {
+		return err
+	}
+
+	defer windows.RegCloseKey(innerKey)
+
+	securityDescriptor, err := winio.SddlToSecurityDescriptor("D:PARAI(A;CIOI;KA;;;BA)(A;CIOI;KA;;;SY)")
+	if err != nil {
+		return err
+	}
+
+	err = system.RegSetKeySecurity(innerKey, system.DACL_SECURITY_INFORMATION, &securityDescriptor[0])
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getComputerSid(accountKey windows.Handle) (string, error) {
+
+	var dataLength uint32
+	var valueType uint32
+	var dataBuffer []byte
+
+	dataLength = 0
+
+	err := windows.RegQueryValueEx(accountKey, windows.StringToUTF16Ptr("V"), nil, &valueType, nil, &dataLength)
+	if err != nil {
+		return "", err
+	}
+
+	if valueType != windows.REG_BINARY {
+		return "", system.ERROR_UNSUPPORTED_TYPE
+	}
+
+	dataBuffer = make([]byte, dataLength)
+
+	err = windows.RegQueryValueEx(accountKey, windows.StringToUTF16Ptr("V"), nil, &valueType, (*byte)(unsafe.Pointer(&dataBuffer[0])), &dataLength)
+	if err != nil {
+		return "", err
+	}
+
+	firstSubAuthority := *(*uint32)(unsafe.Pointer(&dataBuffer[dataLength-12]))
+	secondSubAuthority := *(*uint32)(unsafe.Pointer(&dataBuffer[dataLength-8]))
+	thirdSubAuthority := *(*uint32)(unsafe.Pointer(&dataBuffer[dataLength-4]))
```

Comment:
I know it's unlikely to receive invalid data here, but what if the length is less than 12, or 8, or 4? It might be best to ensure some minimum required length before dereferencing with pointer arithmetic.